vpd-tool/src/vpd_tool.cpp

```diff
@@ -0,0 +1,53 @@
+#include "vpd_tool.hpp"
+
+#include "tool_constants.hpp"
+#include "utils.hpp"
+
+#include <iostream>
+
+namespace vpd
+{
+void VpdTool::readKeyword(const std::string& i_fruPath,
+                          const std::string& i_recordName,
+                          const std::string& i_keywordName, bool i_onHardware,
+                          const std::string& i_fileToSave)
+{
+    types::DbusVariantType l_keywordValue;
+    if (i_onHardware)
+    {
+        std::cout << "Read from h/w" << std::endl;
+        // TODO: uncomment when implementation of the API goes in
+        /*l_keywordValue = utils::readKeywordFromHardware(
+            tool_constants::vpdManagerService,
+           tool_constants::vpdManagerObjectPath,
+            tool_constants::vpdManagerIntfName, i_fruPath,
+            make_tuple(i_recordName, i_keywordName));*/
+    }
+    else
+    {
+        std::string l_fullObjectPath(tool_constants::baseInventoryPath +
+                                     i_fruPath);
+
+        l_keywordValue = utils::readDbusProperty(
+            tool_constants::inventoryManagerService, l_fullObjectPath,
+            "com.ibm.ipzvpd." + i_recordName, i_keywordName);
+    }
+
+    if (const auto l_value = std::get_if<types::BinaryVector>(&l_keywordValue))
```

Comment:
is the reading restricted to vector of uint8_t? If so doxygen does not tell that. Also in case of any other type log at the end about invalid data type received.

Reply:
Added note in doxygen.
 and updated with else for logging